### PR TITLE
Fix auth bug

### PR DESCRIPTION
**Pull request by andela-sjames**
#### What Does This PR Do?
Fix authentication bug for Non-Andela access to application
#### Any Background Context You Want To Provide?
N/A

#### Screenshot(s)
N/A

#### Any new frontend dependency?
N/A

#### Any new backend dependency?
N/A

### CODE REVIEW

File: reactlibapp/client/src/reducers/authReducer.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export default function authReducer(state: object = {}, action: IAction = { type
     case auth.SIGNIN_FAILURE:
     case auth.SIGNOUT_SUCCESS:
       return {
+        ...action.payload,
```
Comment (andela-pessien) [inline reactlibapp/client/src/reducers/authReducer.ts]: Do we intend to spread the error object at the top level of the state?

Comment (andela-sjames) [reply reactlibapp/client/src/reducers/authReducer.ts]: Yes, it's a simple object. 
error

Comment (andela-sjames) [reply reactlibapp/client/src/reducers/authReducer.ts]: @andela-pessien

Comment (andela-sjames) [reply reactlibapp/client/src/reducers/authReducer.ts]: This way message from the server can be used directly on the client to tell the user what the actual error is as we don't have any from what I can see.

Comment (andela-pessien) [reply reactlibapp/client/src/reducers/authReducer.ts]: I get that, it's more of the scope of the state

`detail` is an error message, but is in the top level of the auth state tree with nothing that indicates that it's an error message.

Comment (andela-sjames) [reply reactlibapp/client/src/reducers/authReducer.ts]: I see what you mean there, 
new_err

I have modified it, with a proper name, the top level of the auth state tree is still fine though. 
@andela-pessien

File: reactlibapp/client/src/types/index.d.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 export interface IState {
   auth: {
     isAuthenticated?: boolean,
```
Comment (BolajiOlajide) [inline reactlibapp/client/src/types/index.d.ts]: I'm not sure but is this meant to be optional.
Looks to me like it shouldn't be, it is what one should be using to check if a user is authenticated or not.

Comment (andela-sjames) [reply reactlibapp/client/src/types/index.d.ts]: Oh, that makes sense... will push a fix

**Comment by BolajiOlajide** [comment]
Great work here @andela-sjames

**Pull request merged**